plugins/destination/bigquery/client/types.go

```diff
@@ -2,46 +2,208 @@ package client
 
 import (
 	"cloud.google.com/go/bigquery"
-	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/apache/arrow/go/v13/arrow"
+	"github.com/cloudquery/plugin-sdk/v3/schema"
+	"github.com/cloudquery/plugin-sdk/v3/types"
 )
 
-func (*Client) SchemaTypeToBigQuery(t schema.ValueType) (ft bigquery.FieldType, repeated bool) {
-	switch t {
-	case schema.TypeBool:
-		return bigquery.BooleanFieldType, false
-	case schema.TypeInt:
-		return bigquery.IntegerFieldType, false
-	case schema.TypeFloat:
-		return bigquery.FloatFieldType, false
-	case schema.TypeUUID:
-		return bigquery.StringFieldType, false
-	case schema.TypeString:
-		return bigquery.StringFieldType, false
-	case schema.TypeByteArray:
-		return bigquery.BytesFieldType, false
-	case schema.TypeStringArray:
-		return bigquery.StringFieldType, true
-	case schema.TypeTimestamp:
-		return bigquery.TimestampFieldType, false
-	case schema.TypeJSON:
-		return bigquery.JSONFieldType, false
-	case schema.TypeUUIDArray:
-		return bigquery.StringFieldType, true
-	case schema.TypeCIDR:
-		return bigquery.StringFieldType, false
-	case schema.TypeCIDRArray:
-		return bigquery.StringFieldType, true
-	case schema.TypeMacAddr:
-		return bigquery.StringFieldType, false
-	case schema.TypeMacAddrArray:
-		return bigquery.StringFieldType, true
-	case schema.TypeInet:
-		return bigquery.StringFieldType, false
-	case schema.TypeInetArray:
-		return bigquery.StringFieldType, true
-	case schema.TypeIntArray:
-		return bigquery.IntegerFieldType, true
+// TimestampNanoseconds is a struct to hold a timestamp with nanosecond precision,
+// because BigQuery does not support nanosecond precision timestamps.
+type TimestampNanoseconds struct {
+	Timestamp   string `json:"timestamp" bigquery:"timestamp"`
+	Nanoseconds int    `json:"nanoseconds" bigquery:"nanoseconds"`
+}
+
+// DurationNanoseconds is a struct to hold a duration with nanosecond precision,
+// because BigQuery does not support nanosecond precision durations.
+type DurationNanoseconds struct {
+	Duration    string `json:"duration" bigquery:"duration"`
+	Nanoseconds int    `json:"nanoseconds" bigquery:"nanoseconds"`
+}
+
+func (c *Client) ColumnToBigQuerySchema(col schema.Column) *bigquery.FieldSchema {
+	sc := bigquery.FieldSchema{
+		Name:        col.Name,
+		Description: col.Description,
+		Type:        c.DataTypeToBigQueryType(col.Type),
+		Schema:      c.DataTypeToBigQuerySchema(col.Type),
+	}
+
+	// TODO: handle repeated; we currently don't handle the case where we get a list of lists,
+	//       but that's not a valid case right now and isn't being explicitly tested for.
+	if col.Type.ID() == arrow.LIST {
+		sc.Repeated = true
+	}
+	return &sc
+}
+
+func (c *Client) DataTypeToBigQueryType(dataType arrow.DataType) bigquery.FieldType {
+	// handle complex types
+	switch dataType.ID() {
+	case arrow.MAP:
+		return bigquery.JSONFieldType
+	case arrow.STRUCT:
+		return bigquery.RecordFieldType
+	case arrow.LIST:
```

Comment:
fixed size list?

Reply:
Hmm, we should add a test for it if we want to support it.

Reply:
I've added some handling, but since we're not testing it I think officially we should say it's not supported yet